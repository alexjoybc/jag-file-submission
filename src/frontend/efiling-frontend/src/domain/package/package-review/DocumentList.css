.header {
  display: flex;
  flex-wrap: wrap;
  font-weight: bold;
}
.label {
  font-weight: bold;
  margin-bottom: 0px;
}
ul {
  margin: 5px 0px;
  padding: 0px;
}
li {
  display: flex;
  flex-wrap: wrap;
  list-style-type: none;
  border: 2px solid #e1bebd;
  padding: 10px 0px;
  margin: 10px 0px;
}
li span {
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}
