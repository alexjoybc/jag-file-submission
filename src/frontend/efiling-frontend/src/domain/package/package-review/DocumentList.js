import React from "react";
import PropTypes from "prop-types";

import "./DocumentList.css";

export default function DocumentList({ documents }) {
  return (
    <div className="document-list">
      <div className="header">
        <span className="d-none d-lg-inline col-lg-3">Document Type</span>
        <span className="d-none d-lg-inline col-lg-4">File Name</span>
        <span className="d-none d-lg-inline col-lg-3">Status</span>
        <span className="d-none d-lg-inline col-lg-2">Action (s)</span>
      </div>
      <ul>
        {documents &&
          documents.map((document) => (
            <li key={document.identifier}>
              <span className="label col-sm-4 d-lg-none">Document Type:</span>
              <span className="col-sm-8 col-lg-3">{document.description}</span>
              <span className="label col-sm-4 d-lg-none">File Name:</span>
              <span className="col-sm-8 col-lg-4">{document.name}</span>
              <span className="label col-sm-4 d-lg-none">Status:</span>
              <span className="col-sm-8 col-lg-3">
                {document.status.description}
              </span>
              <span className="label col-sm-4 d-lg-none">Action (s):</span>
              <span className="col-sm-8 col-lg-2">withdraw</span>
            </li>
          ))}
      </ul>
    </div>
  );
}

DocumentList.propTypes = {
  documents: PropTypes.arrayOf(
    PropTypes.shape({
      identifier: PropTypes.string.isRequired,
      type: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      status: PropTypes.shape({
        description: PropTypes.string.isRequired,
      }),
    })
  ).isRequired,
};
