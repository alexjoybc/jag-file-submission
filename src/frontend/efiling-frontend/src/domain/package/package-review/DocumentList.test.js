import React from "react";
import { render, waitFor, fireEvent, getByText } from "@testing-library/react";
import DocumentList from "./DocumentList";

const pkgRvwService = require("./PackageReviewService");

describe("DocumentList Component Testsuite", () => {
  const packageId = "1";
  const documents = [
    {
      identifier: "1",
      type: "AFF",
      name: "test-document.pdf",
      status: {
        description: "Submitted",
      },
    },
  ];

  beforeEach(() => {
    window.open = jest.fn();
  });

  test("Matches the snapshot", async () => {
    const { asFragment } = render(
      <DocumentList packageId={packageId} documents={documents} />
    );
    await waitFor(() => {});
    expect(asFragment()).toMatchSnapshot();
  });

  test("Download document (click) success", async () => {
    // stub out service to return valid response.
    pkgRvwService.downloadSubmittedDocument = jest.fn(() => Promise.resolve());

    const { container } = render(
      <DocumentList packageId={packageId} documents={documents} />
    );
    await waitFor(() => {});

    // get the span wrapping the file link, click it.
    fireEvent.click(getByText(container, `${documents[0].name}`));
    await waitFor(() => {});

    expect.assertions(1);
    expect(pkgRvwService.downloadSubmittedDocument()).resolves.not.toThrow();
  });

  test("Download document (click) fail", async () => {
    const errorMessage = "Network Error";
    // stub out service to return failed response.
    pkgRvwService.downloadSubmittedDocument = jest.fn(() =>
      Promise.reject(new Error(errorMessage))
    );

    const { container } = render(
      <DocumentList packageId={packageId} documents={documents} />
    );
    await waitFor(() => {});

    // get the span wrapping the file link, click it.
    fireEvent.click(getByText(container, `${documents[0].name}`));
    await waitFor(() => {});

    expect.assertions(1);
    expect(pkgRvwService.downloadSubmittedDocument()).rejects.toThrow(
      errorMessage
    );
  });

  test("Download document (keyDown) success keyCode=13", async () => {
    // stub out service to return valid response.
    pkgRvwService.downloadSubmittedDocument = jest.fn(() => Promise.resolve());

    const { container } = render(
      <DocumentList packageId={packageId} documents={documents} />
    );
    await waitFor(() => {});

    // get the span wrapping the file link, click it.
    fireEvent.keyDown(getByText(container, `${documents[0].name}`), {
      key: "Enter",
      keyCode: "13",
    });
    await waitFor(() => {});

    expect.assertions(1);
    expect(pkgRvwService.downloadSubmittedDocument()).resolves.not.toThrow();
  });

  test("Download document (keyDown) success keyCode=32", async () => {
    // stub out service to return valid response.
    pkgRvwService.downloadSubmittedDocument = jest.fn(() => Promise.resolve());

    const { container } = render(
      <DocumentList packageId={packageId} documents={documents} />
    );
    await waitFor(() => {});

    // get the span wrapping the file link, click it.
    fireEvent.keyDown(getByText(container, `${documents[0].name}`), {
      key: "Space",
      keyCode: "32",
    });
    await waitFor(() => {});

    expect.assertions(1);
    expect(pkgRvwService.downloadSubmittedDocument).not.toHaveBeenCalled();
  });

  test("Download document (click) fail", async () => {
    const errorMessage = "Network Error";
    // stub out service to return failed response.
    pkgRvwService.downloadSubmittedDocument = jest.fn(() =>
      Promise.reject(new Error(errorMessage))
    );

    const { container } = render(
      <DocumentList packageId={packageId} documents={documents} />
    );
    await waitFor(() => {});

    // get the span wrapping the file link, click it.
    fireEvent.keyDown(getByText(container, `${documents[0].name}`), {
      key: "Enter",
      keyCode: "13",
    });
    await waitFor(() => {});

    expect.assertions(1);
    expect(pkgRvwService.downloadSubmittedDocument()).rejects.toThrow(
      errorMessage
    );
  });
});
