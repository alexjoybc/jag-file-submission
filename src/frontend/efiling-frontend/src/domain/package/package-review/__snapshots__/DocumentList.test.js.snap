// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentList Component Testsuite Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="document-list"
  >
    <div
      class="header"
    >
      <span
        class="d-none d-lg-inline col-lg-3"
      >
        Document Type
      </span>
      <span
        class="d-none d-lg-inline col-lg-4"
      >
        File Name
      </span>
      <span
        class="d-none d-lg-inline col-lg-3"
      >
        Status
      </span>
      <span
        class="d-none d-lg-inline col-lg-2"
      >
        Action (s)
      </span>
    </div>
    <ul>
      <li>
        <span
          class="label col-sm-4 d-lg-none"
        >
          Document Type:
        </span>
        <span
          class="col-sm-8 col-lg-3"
        />
        <span
          class="label col-sm-4 d-lg-none"
        >
          File Name:
        </span>
        <span
          class="col-sm-8 col-lg-4 file-cell"
        >
          <span
            class="file-href"
            role="button"
            tabindex="0"
          >
            test-document.pdf
          </span>
        </span>
        <span
          class="label col-sm-4 d-lg-none"
        >
          Status:
        </span>
        <span
          class="col-sm-8 col-lg-3"
        >
          Submitted
        </span>
        <span
          class="label col-sm-4 d-lg-none"
        >
          Action (s):
        </span>
        <span
          class="col-sm-8 col-lg-2"
        >
          withdraw
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
